src/Toolkit/Toolkit/UI/Controls/SearchView/LocatorSearchSource.cs

```diff
@@ -146,7 +146,19 @@ public LocatorSearchSource(LocatorTask locator)
         {
             Locator = locator;
 
-            _loadTask = EnsureLoaded();
+            RefreshDisplayName();
+
+            _loadTask = new Lazy<Task>(EnsureLoaded);
+        }
+
+        private void RefreshDisplayName()
+        {
+            if (DisplayName != Locator?.LocatorInfo?.Name && !string.IsNullOrWhiteSpace(Locator?.LocatorInfo?.Name))
+            {
+                DisplayName = Locator?.LocatorInfo?.Name ?? "Locator";
+            }
+
+            GeocodeParameters.ResultAttributeNames.Add("*");
```

Comment:
Shouldn't this need to be called again once the load task completes and then raise property changed for the displayname property? Also "Locator" should be localized, but it's probably better to leave it empty, and then update it once the task completes.
I would think that hitting the getter for DisplayName should also initiate the load and return empty (and then raise INPC once it completes).